src/libraries/System.Net.Http/tests/FunctionalTests/HttpWebtransportSessionTest.cs

```diff
@@ -0,0 +1,881 @@
+
+        [Theory]
+        [InlineData(QuicStreamType.Unidirectional, true)]
+        [InlineData(QuicStreamType.Unidirectional, false)]
+        [InlineData(QuicStreamType.Bidirectional, true)]
+        [InlineData(QuicStreamType.Bidirectional, false)]
+        public async Task WebtransportStreamMultipleReadsAndWrites(QuicStreamType type, bool clientInitiated)
+        {
+            using Http3LoopbackServer server = CreateHttp3LoopbackServer();
+            byte[] s_data = "Hello world!"u8.ToArray();
+            const int sendCount = 5;
+            int expectedBytesCount = s_data.Length * sendCount;
+            byte[] expected = new byte[expectedBytesCount];
+            Memory<byte> m = expected;
+            for (int i = 0; i < sendCount; i++)
+            {
+                s_data.CopyTo(m);
+                m = m[s_data.Length..];
+            }
+            Task serverTask = Task.Run(async () =>
+            {
+                ICollection<(long settingId, long settingValue)> settings = new LinkedList<(long settingId, long settingValue)>();
+                settings.Add((Http3LoopbackStream.EnableWebTransport, 1));
+                var headers = new List<HttpHeaderData>();
+                int contentLength = 10;
+                HttpHeaderData header = new HttpHeaderData("sec-webtransport-http3-draft", "draft02");
+                headers.Add(new HttpHeaderData("Content-Length", contentLength.ToString(CultureInfo.InvariantCulture)));
+                headers.Add(header);
+
+                await using Http3LoopbackConnection connection = (Http3LoopbackConnection)await server.EstablishGenericConnectionAsync(settings);
+                Http3LoopbackStream stream = await connection.AcceptRequestStreamAsync();
+                await stream.ReadRequestDataAsync(false);
+                await stream.SendResponseAsync(HttpStatusCode.OK, headers, "", false);
+                QuicStream help;
+                if(!clientInitiated)
+                {
+                    help = await connection.OpenOutboundStreamAsync(type);
+                    await SendWebtransportStreamHeaderAsync(help, (type == QuicStreamType.Unidirectional) ?
+                        Http3LoopbackStream.UnidirectionalWebtransportStream : Http3LoopbackStream.BidirectionalWebtransportStream, stream.StreamId);
+                }
+                else
+                {
+                    help = await connection.AcceptWebtransportStreamAsync();
+                    (long? frameType, long? sessionId) = await ReadWTFrameAsync(help);
+                    Assert.Equal(stream.StreamId, sessionId);
+
+                }
+
+                if (!clientInitiated || (clientInitiated && type == QuicStreamType.Bidirectional))
+                {
+                    for (int i = 0; i < sendCount; i++)
+                    {
+                        await help.WriteAsync(s_data);
+                    }
+                    await help.WriteAsync(Memory<byte>.Empty, completeWrites: true);
```

Comment:
you can use `CompleteWrites()` instead (it's doing the same thing)

(why is it called "help"? 🙂)